docs/dev/Pagination-v2.md

```diff
@@ -100,10 +111,381 @@ The discussion below uses *under max_result_window* to refer to scenarios that c
 ## SQL Node Load Balancing
 V2 SQL engine supports *sql node load balancing* -- a cursor request can be routed to any SQL node in a cluster. This is achieved by encoding all data necessary to retrieve the next page in the `cursor_id`.
 
-## Design Diagrams
-New code workflows are highlighted.
+## Feature Design
+
+### Plan Tree changes
+
+There are different plan trees are built during request processing. Read more about their purpose and stages [here](query-optimizer-improvement.md#Examples). Article below describes what changes are introduced in these trees by pagination feature.
+
+Simplified workflow of plan trees is shown below. Initial Page Request is processed the same way as non-paging request.
```

Comment:
```suggestion
Simplified workflow of plan trees is shown below. Initial Page Request is processed the same way as a non-paging request.
```

Reply:
> Initial Page Request is processed the same way as non-paging request
Is this true?  Why give us two diagrams for "Non Paged Request" and "Initial Page Request"?  Why not just combine them into a single state diagram?